### PR TITLE
Encapsulate `Url` struct creation in constructor function

### PR DESCRIPTION
**Pull request by 18a6**

### CODE REVIEW

**Comment by achab** [comment]
@18a6 some changes shouldn't appear on this branch since they are already on main. please read about git rebase and run the following commands:
  ```bash
  git checkout main
  git pull origin main --rebase
  git checkout feat/go-project
  git rebase main  # here you are rebasing your branch on main, expect conflicts
  git push origin feat/go-project --force  # here we add the force option because we want to override git history
  ```

**Comment by 18a6** [comment]
I resolved conflicts. Is it ok now?

**Comment by achab** [comment]
> I resolved conflicts. Is it ok now?

If you do a git rebase, you will have to fix the conflicts locally.

File: extract.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+// returns the Ether balance of a given address
+func getBalance(address, tag string) string {
+	return call(buildUrl("balance", address, tag))
+}
+
+func getBalances(address, tag string) string {
+	return call(buildUrl("balancemulti", address, tag))
+}
+
+func main() {
```
Comment (achab) [inline extract.go:67]: can you add a comment to explain what this contract at `0xde0b295669a9fd93d5f28d9ec85e40f4cb697bae` does?

Comment (achab) [reply extract.go:67]: ok got it. regarding conflicts, this is indeed annoying the fix all of them, but this is how you learn how to better work with git.

Comment (18a6) [reply extract.go:67]: What if I care only about the last commit, can I squash commits into one?

Comment (achab) [reply extract.go:67]: Yes you can, but you just focus on rebasing first. We can squash commits as a next step.

Comment (achab) [reply extract.go:67]: Anyway, commits will be squashed when you merge your PR to main.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,28 @@
...
-navigate to the cloned folder, then run: `go run extract.go`
\ No newline at end of file
+## Instructions to run
+
+clone the repo: `git clone https://github.com/Amenokal-Labs/mev-mantis.git`.
+
+navigate to the cloned folder, then run: `go run extract.go`.
+
+## Docs
+<<<<<<< HEAD
```
Comment (achab) [inline README.md:10]: please remove lines starting with `<<<<<<<` and `>>>>>>>`. they come from the rebase process

File: extract.go (hunk 1 above)
Comment (achab) [reply extract.go:68]: :rocket:

**Pull request merged**